base/sysimg.jl

```diff
@@ -1,6 +1,12 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-Core.include(Main, "Base.jl")
+# Can be built either a monolith or with a minimal Base image that just has the
+# compiler.
+if isdefined(Main, :Base); else
+Core.eval(Main, :(baremodule Base; end))
+end
```

Comment:
Why this weird way to have three different ways that all appear to do the same thing (in particular, that all define Compiler, while not actually successfully appearing to build a Base without it)? Why not just have 2 files to do this instead of all these weird conditional, like the trivial base/compilerimg.jl, but adding one extra line there instead whether it should stop after loading the compiler or continue to loading sysimg?

Reply:
The objective is to retain the ability to modify `Base` without rebuilding the compiler because the compiler takes significantly longer to bootstrap than Base itself. `compilerimg.jl` and `Base_compiler.jl` could probably be merged.

Reply:
Yeah, that seems fine to have the ability to split the build in half, just doesn't seem useful to have this many different ways to do it based on isdefined tests (rather than just a few different top files that define the different possible load orders)

Reply:
There's really just supposed to be two (split built and full monolith), but I can merge `compilerimg.jl` and `Base_compiler.jl` and delete the conditionals.